### PR TITLE
X-ray flash protection changes

### PR DESCRIPTION
**Pull request by BrumalLake**
## About The Pull Request
Having x-ray now ignores any flash protection you have, excluding shield eyes if you are using x-ray goggles
Uplink autosurgeons come with a variant of x-ray eyes that can still take flash protection
You do not get any tint from clothing while wearing x-ray, as you can see through it
Flashbangs also can flash x-ray users through walls

## Why It's Good For The Game
X-ray is a straight upgrade from thermal vision

## Changelog
:cl:
balance: having x-ray makes you ignore flash protection and tint
balance: flashbangs can flash people with x-ray through walls
add: syndicate autosurgeons get a special variant of eyes that allow flash protection
/:cl:

### CODE REVIEW

**Comment by BrumalLake** [comment]
I think people werent flashable through glass before either and now they are

File: code/game/objects/items/grenades/flashbang.dm [hunk 1]
```diff
@@ -30,6 +30,8 @@
 		M.Paralyze(max(20/max(1,distance), 5))
 		M.Knockdown(max(200/max(1,distance), 60))
 //Bang
+	if(!(M in hearers(flashbang_range, get_turf(src))))
```
Comment (PowerfulBacon) [inline code/game/objects/items/grenades/flashbang.dm]: This is super inefficient, you are doing hearers not 1 time now but 1 + n times where n is the number of mobs that can see the flashbang

File: code/modules/clothing/glasses/_glasses.dm [hunk 2]
```diff
@@ -306,15 +306,6 @@
...
 
-/obj/item/clothing/glasses/blindfold/equipped(mob/living/carbon/human/user, slot)
-	. = ..()
-	if(slot == ITEM_SLOT_EYES)
-		user.become_blind("blindfold_[REF(src)]")
-
-/obj/item/clothing/glasses/blindfold/dropped(mob/living/carbon/human/user)
-	..()
-	user.cure_blind("blindfold_[REF(src)]")
-
```
Comment (PowerfulBacon) [inline code/modules/clothing/glasses/_glasses.dm:322]: ?

File: code/modules/mob/living/carbon/carbon.dm [hunk 3]
```diff
@@ -615,6 +615,9 @@
 	if(!GLOB.tinted_weldhelh)
 		return
 	tinttotal = get_total_tint()
+	var/obj/item/organ/eyes/E = getorganslot(ORGAN_SLOT_EYES)
+	if(E && !(~sight & (SEE_MOBS|SEE_OBJS|SEE_TURFS)))
+		return
```
Comment (PowerfulBacon) [inline code/modules/mob/living/carbon/carbon.dm]: This makes sense but the implementation is poorly coded

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 4]
```diff
@@ -4,6 +4,8 @@
 	var/obj/item/organ/eyes/E = getorganslot(ORGAN_SLOT_EYES)
 	if(!E)
 		return INFINITY //Can't get flashed without eyes
+	if(!(~sight & (SEE_MOBS|SEE_OBJS|SEE_TURFS)) && !(istype(E, /obj/item/organ/eyes/robotic/xray/syndicate) || istype(E, /obj/item/organ/eyes/robotic/shield)))
+		return 0
```
Comment (PowerfulBacon) [inline code/modules/mob/living/carbon/carbon_defense.dm]: Poor code quality, should be a var that protects it and checking the sight vars is pretty dodgy imo

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]

File: code/modules/clothing/glasses/_glasses.dm (hunk 2 above)
Comment (BrumalLake) [reply code/modules/clothing/glasses/_glasses.dm:322]: its useless, the 3 tint on it already makes you blind and im not gonna put logic for x ray flags

File: code/modules/mob/living/carbon/carbon.dm (hunk 3 above)
Comment (BrumalLake) [reply code/modules/mob/living/carbon/carbon.dm]: whats wrong with it

File: code/modules/mob/living/carbon/carbon_defense.dm (hunk 4 above)
Comment (BrumalLake) [reply code/modules/mob/living/carbon/carbon_defense.dm]: you can't get those 3 flags at the same time from anything other than x-ray

**Comment by utilbird** [comment]
Shouldn't thermal vision be *more* affected by this

**Comment by BrumalLake** [comment]
> 
> 
> Shouldn't thermal vision be _more_ affected by this

what do you mean

**Comment by Isy232** [comment]
> Shouldn't thermal vision be _more_ affected by this

I dont think our thermals are FLIR or IRNV or any sort of realistic thing

They're magical bullshit because they can see heat through solid objects.

Thermals are already kinda bad in comparison to x-ray, so I would think just let it be like this.

File: code/modules/mob/living/carbon/carbon_defense.dm (hunk 4 above)
Comment (PowerfulBacon) [reply code/modules/mob/living/carbon/carbon_defense.dm]: That doesn't mean they could be added in the future and that it's not dodgy code

File: code/modules/mob/living/carbon/carbon.dm (hunk 3 above)
Comment (PowerfulBacon) [reply code/modules/mob/living/carbon/carbon.dm]: Checking specifically to see the eye flags rather than using / adding vars that affect flash protection

**Review by Archanial — DISMISSED** [review]

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
Literally just give x-ray eyes -5 flash protection or whatever (over the dodgy code being done with checking the sight flash of the mob for a specific set of vision flags only used by the eyes.)

**Comment by lordScrubling** [comment]
> 
> 
> Literally just give x-ray eyes -5 flash protection or whatever (over the dodgy code being done with checking the sight flash of the mob for a specific set of vision flags only used by the eyes.)

that won't bode well for having flash protective eyes under x-ray goggles which are syndicate only

**Comment by BrumalLake** [comment]
> 
> 
> Literally just give x-ray eyes -5 flash protection or whatever (over the dodgy code being done with checking the sight flash of the mob for a specific set of vision flags only used by the eyes.)

one thing i wish i could do was remove the -flash on thermals since welding seems to be getting buffed to blind even if you have 1 flash protection so you won't be able to weld even with a welding screen down with thermals

**Review by qwertyquerty — APPROVED** [review]

**Pull request merged**